runtime/parachains/src/disputes/migration.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017-2022 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Storage migration(s) related to disputes pallet
+
+#[allow(deprecated)]
+use crate::disputes::{pallet::SpamSlots, Config, Pallet};
+use frame_support::{
+	pallet_prelude::*,
+	traits::{OnRuntimeUpgrade, StorageVersion},
+	weights::Weight,
+};
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+pub struct MigrateToV1<T>(sp_std::marker::PhantomData<T>);
+impl<T: Config> OnRuntimeUpgrade for MigrateToV1<T> {
+	fn on_runtime_upgrade() -> Weight {
+		let mut weight: Weight = Weight::zero();
+
+		if StorageVersion::get::<Pallet<T>>() < STORAGE_VERSION {
+			weight += migrate_to_v1::<T>();
+			STORAGE_VERSION.put::<Pallet<T>>();
+			weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
+		}
+		weight
+	}
+}
+
+/// Migrates the pallet storage to the most recent version, checking and setting the `StorageVersion`.
+pub fn migrate_to_v1<T: Config>() -> Weight {
+	let weight: Weight = Weight::zero();
+
+	// SpamSlots should not contain too many keys so removing everything at once should be safe
+	loop {
+		#[allow(deprecated)]
+		let res = SpamSlots::<T>::clear(u32::MAX, None);
```

Comment:
If you call this with `u32::MAX`, isn't the cursor guaranteed to be `none`?

Reply:
I would assume so, but the question is, whether it is safe to make it u32::MAX.

Reply:
As suggested in https://github.com/paritytech/polkadot/pull/6345#discussion_r1050877140, we should remove the loop. The initial limit was set to 0 (probably misunderstanding of the API).

My understanding is that there's no risk of creating an overweight block because the way the weight is accounted for in migration is based on number of keys deleted, which is limited by 6 (db reads/writes). So it doesn't take into account the sizes of values. If they grow large, importing the block will just take longer than 6s. But this is fine for one-off migration. Am I missing something?

Reply:
Oh yeah! Indeed, I should have taken a closer look on how this actually looks like. Indeed there can only be 6 entries worst case. So this is totally harmless. :face_in_clouds: :partying_face: :rofl: